archunit/src/test/java/com/tngtech/archunit/lang/syntax/elements/GivenMembersTest.java

```diff
@@ -214,6 +214,23 @@ public void complex_members_syntax() {
                         allConstructorsExcept(CONSTRUCTOR_ONE_ARG)),
                 $(described(fields().that().haveFullNameNotMatching(quote(classNameDot) + ".*A.*")), allFieldsExcept(FIELD_A)),
 
+
+                $(described(members().that().haveNameStartingWith("fie")), ALL_FIELD_DESCRIPTIONS),
+                $(described(codeUnits().that().haveNameStartingWith("me")), ALL_METHOD_DESCRIPTIONS),
+                $(described(methods().that().haveNameStartingWith("m")), ALL_METHOD_DESCRIPTIONS),
+                $(described(constructors().that().haveNameStartingWith("<")), ALL_CONSTRUCTOR_DESCRIPTIONS),
+                $(described(fields().that().haveNameStartingWith("f")), ALL_FIELD_DESCRIPTIONS),
+                $(described(members().that().haveNameContaining("et")), ALL_METHOD_DESCRIPTIONS),
+                $(described(codeUnits().that().haveNameContaining("t")), ALL_CODE_UNIT_DESCRIPTIONS),
+                $(described(methods().that().haveNameContaining("dA")), ImmutableSet.of(METHOD_A)),
+                $(described(constructors().that().haveNameContaining("init")), ALL_CONSTRUCTOR_DESCRIPTIONS),
+                $(described(fields().that().haveNameContaining("ie")), ALL_FIELD_DESCRIPTIONS),
+                $(described(members().that().haveNameEndingWith("D")), ImmutableSet.of(FIELD_D, METHOD_D)),
+                $(described(codeUnits().that().haveNameEndingWith("A")), ImmutableSet.of(METHOD_A)),
+                $(described(methods().that().haveNameEndingWith("C")), ImmutableSet.of(METHOD_C)),
+                $(described(constructors().that().haveNameEndingWith("it>")), ALL_CONSTRUCTOR_DESCRIPTIONS),
+                $(described(fields().that().haveNameEndingWith("B")), ImmutableSet.of(FIELD_B)),
+
```

Comment:
We could probably add similar test coverage for the _members should_ functinality in `MembersShouldTest`.